client/src/featureform/status_display.py

```
@@ -151,6 +155,9 @@ def display(self):
                 live.refresh()
 
                 if finished_running:
+                    if self.did_error:
+                        sys.tracebacklimit = 0
+                        raise Exception("Some resources failed to create")
```

what is this for?

---

Right now we just break when things are done, rather than exit(1). However, if you exit(1) it kills the parent process aka the tests. This format gives a failed exit without a massive stack trace

---

so this is purely for tests?

---

if it's just for tests I would leave a comment or something.

---

and any usage where apply would be synchronous and fail. Otherwise it fails silently

---

like in .dataframe

---

I'd still add a comment on what's going on. I know I'll look at this in a few months about be like wtf is going on